test/sanitize-field-names/express.js

```diff
@@ -0,0 +1,91 @@
+'use strict'
+const {
+  createAgentConfig,
+  resetAgent,
+  assertRequestHeadersWithFixture,
+  assertResponseHeadersWithFixture,
+  assertFormsWithFixture
+} = require('./shared')
+const agent = require('../..').start(createAgentConfig())
+const test = require('tape')
+const request = require('request')
+const express = require('express')
+const bodyParser = require('body-parser')
+const fixtures = require('./fixtures')
+
+function runTest (
+  t, expected, agentConfig, requestHeaders, responseHeaders, formFields, middleware = false
+) {
+  agent._config(agentConfig)
+  const app = express()
+  if (middleware) {
+    app.use(middleware)
+  }
+
+  // resets agent values for tests.  Callback fires
+  // after mockClient receives data
+  resetAgent(agent, (data) => {
+    const transaction = data.transactions.pop()
+    assertRequestHeadersWithFixture(transaction, expected, t)
+    assertResponseHeadersWithFixture(transaction, expected, t)
+    assertFormsWithFixture(transaction, expected, t)
+  })
+
+  // register request handler
+  app.post('/test', (req, res) => {
+    t.ok('received request', 'received request')
+    res.header(responseHeaders)
+    res.send('Hello World')
+  })
+
+  const server = app.listen(0, '0.0.0.0', () => {
+    const url = `http://${server.address().address}:${server.address().port}/test`
+    request.post(
+      url,
+      {
+        form: formFields,
+        headers: requestHeaders
+      },
+      function (error, response, body) {
+        if (error) {
+          t.fail(error)
+        }
+        t.ok(body, 'received response')
+        t.end()
+      })
+  })
+
+  const done = () => {
+    server.close()
+  }
+  t.on('end', done)
+}
+
+function createMiddleware (type) {
+  if (type === 'urlencoded') {
+    return bodyParser.urlencoded({ extended: false })
+  } else if (type === 'text') {
+    return bodyParser.text({ type: '*/*' })
+  } else if (type === 'raw') {
+    return bodyParser.raw({ type: '*/*' })
+  }
+
+  throw new Error(`I don't know how to create a ${type} middleware`)
+}
+
+test('Running fixtures with express', function (t) {
+  for (const [, fixture] of fixtures.entries()) {
+    test(fixture.name, function (t2) {
```

Comment:
Perhaps use [subtest](https://github.com/substack/tape#ttestname-opts-cb) syntax here:

```diff
--- a/test/sanitize-field-names/express.js
+++ b/test/sanitize-field-names/express.js
@@ -73,11 +73,11 @@ function createMiddleware (type) {
   throw new Error(`I don't know how to create a ${type} middleware`)
 }

-test('Running fixtures with express', function (t) {
+test('Running fixtures with express', function (suite) {
   for (const [, fixture] of fixtures.entries()) {
-    test(fixture.name, function (t2) {
+    suite.test(fixture.name, function (t) {
       runTest(
-        t2,
+        t,
         fixture.expected,
         createAgentConfig(fixture.agentConfig),
         fixture.input.requestHeaders,
@@ -87,5 +87,5 @@ test('Running fixtures with express', function (t) {
       )
     })
   }
-  t.end()
+  suite.end()
 })
```

I'm not married to "suite", but it is the name I've tended to use for a wrapper with subtests.  Because there are no subsequent top-level `test(...)` usages it probably doesn't make a difference here, but in general it can make a diff.

Reply:
I like it -- `suite`'s a good name for it.  I've updated these tests as well as the similar invocation in the other files.